Mage.Sets/src/mage/cards/j/JaredCarthalion.java

```diff
@@ -0,0 +1,135 @@
+import mage.game.permanent.token.TreasureToken;
+import mage.players.Player;
+import mage.target.common.TargetCardInYourGraveyard;
+import mage.target.common.TargetCreaturePermanent;
+
+import java.util.UUID;
+
+/**
+ * @author PurpleCrowbar
+ */
+public final class JaredCarthalion extends CardImpl {
+
+    private static final FilterCard filter = new FilterCard("multicolored card from your graveyard");
+
+    static {
+        filter.add(MulticoloredPredicate.instance);
+    }
+
+    public JaredCarthalion(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.PLANESWALKER}, "{W}{U}{B}{R}{G}");
+        this.addSuperType(SuperType.LEGENDARY);
+        this.subtype.add(SubType.JARED);
+
+        this.setStartingLoyalty(5);
+
+        // +1: Create a 3/3 Kavu creature token with trample that's all colors.
+        this.addAbility(new LoyaltyAbility(new CreateTokenEffect(new KavuAllColorToken()), 1));
+
+        // −3: Choose up to two target creatures. For each of them, put a number of +1/+1 counters on it equal to the number of colors it is.
+        Ability ability1 = new LoyaltyAbility(new JaredCarthalionBoostEffect(), -3);
+        ability1.addTarget(new TargetCreaturePermanent(0, 2));
+        this.addAbility(ability1);
+
+        // −6: Return target multicolored card from your graveyard to your hand. If that card was all colors, draw a card and create two Treasure tokens.
+        Ability ability2 = new LoyaltyAbility(new JaredCarthalionUltimateEffect(), -6);
+        ability2.addTarget(new TargetCardInYourGraveyard(filter));
+        this.addAbility(ability2);
+
+        // Jared Carthalion can be your commander.
+        this.addAbility(CanBeYourCommanderAbility.getInstance());
+    }
+
+    private JaredCarthalion(final JaredCarthalion card) {
+        super(card);
+    }
+
+    @Override
+    public JaredCarthalion copy() {
+        return new JaredCarthalion(this);
+    }
+}
+
+class JaredCarthalionBoostEffect extends OneShotEffect {
+
+    public JaredCarthalionBoostEffect() {
+        super(Outcome.BoostCreature);
+        this.staticText = "Choose up to two target creatures. For each of them, put " +
+                "a number of +1/+1 counters on it equal to the number of colors it is.";
+    }
+
+    public JaredCarthalionBoostEffect(final JaredCarthalionBoostEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public JaredCarthalionBoostEffect copy() {
+        return new JaredCarthalionBoostEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        for (UUID targetId : getTargetPointer().getTargets(game, source)) {
+            Permanent creature = game.getPermanent(targetId);
+            if (creature != null) {
+                Counter counter;
+                counter = CounterType.P1P1.createInstance(creature.getColor().getColorCount());
+                creature.addCounters(counter, source, game);
```

Comment:
you don’t need to declare the counter variable on a separate line, also you should get the color count, check if it’s positive, and then add the counters

Reply:
Is it possible for a card to have a negative number of colors?

Reply:
no but it can have zero

Reply:
In testing with a colorless creature (Solemn Simulacrum), the effect resolves as expected without error and without the creature receiving any counters. Do you think I should change it anyway or?

Reply:
it should work either way, I'd just rather short circuit creating the counters and calling the method